worker/worker/ffmegtools.go

```diff
@@ -14,12 +16,42 @@ func getDuration(file string) (float64, error) {
 	return gjson.Get(probe, "format.duration").Float(), nil
 }
 
-func getCodec(file string) (string, error) {
+func getVideoCodec(file string) (string, error) {
 	probe, err := probe(file)
 	if err != nil {
 		return "", err
 	}
-	return gjson.Get(probe, "streams.0.codec_name").String(), nil
+	codecNumber := gjson.Get(probe, "streams.#").Int()
+	videoIndex := -1
+	for i := 0; i < int(codecNumber); i++ {
+		if gjson.Get(probe, fmt.Sprintf("streams.%d.codec_type", i)).String() == "video" {
+			videoIndex = int(gjson.Get(probe, fmt.Sprintf("streams.%d.index", i)).Int())
+			break
+		}
+	}
+	if videoIndex != -1 {
+		return gjson.Get(probe, fmt.Sprintf("streams.%d.codec_name", videoIndex)).String(), nil
+	}
+	return "", errors.New("no video stream found")
+}
+
+func getAudioCodec(file string) (string, error) {
```

Comment:
I wonder if we could combine these methods into one, something like this:
```suggestion
// getCodec returns the codec of the file, codecType=video returns the video codec, codecType=audio the audio codec.
func getCodec(file string, codecType string) (string, error) {
```